x-pack/plugins/ingest_manager/server/services/epm/packages/assets.ts

```diff
@@ -57,8 +57,10 @@ export async function getAssetsData(
   datasetName?: string
 ): Promise<Registry.ArchiveEntry[]> {
   // TODO: Needs to be called to fill the cache but should not be required
-  const pkgkey = packageInfo.name + '-' + packageInfo.version;
-  if (!cacheHas(pkgkey)) await Registry.getArchiveInfo(packageInfo.name, packageInfo.version);
+  const pkgkey = getCacheKey(packageInfo.name + '-' + packageInfo.version);
```

Comment:
Haven't tested locally, but how about moving the two instances of this pattern to something like `getOrFetchArchiveInfo` like https://github.com/neptunian/kibana/blob/944540f09cf0c0c33e17273f8bb9ab4c83900b5f/x-pack/plugins/ingest_manager/server/services/epm/registry/index.ts#L136

If it returns the Promise I think we can `await getOrFetchArchiveInfo(...)`